recipes/openblas/all/conanfile.py

```diff
@@ -19,134 +23,214 @@ class OpenblasConan(ConanFile):
+
+    def build_requirements(self):
+        if self.options.build_lapack and self.settings.os == "Windows":
+            self.tool_requires("ninja/1.11.1")
+
     def source(self):
-        tools.get(
-            **self.conan_data["sources"][self.version],
-            strip_root=True,
-            destination=self._source_subfolder
-        )
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
 
-    @functools.lru_cache(1)
-    def _configure_cmake(self):
-        cmake = CMake(self)
+    def generate(self):
 
+        ninja_generator = "Ninja" if self.settings.os == "Windows" and \
+            self.options.build_lapack else None
+        tc = CMakeToolchain(self, generator=ninja_generator)
+
+        tc.variables["NOFORTRAN"] = not self.options.build_lapack
         if self.options.build_lapack:
-            self.output.warn("Building with lapack support requires a Fortran compiler.")
-        cmake.definitions["NOFORTRAN"] = not self.options.build_lapack
-        cmake.definitions["BUILD_WITHOUT_LAPACK"] = not self.options.build_lapack
-        cmake.definitions["DYNAMIC_ARCH"] = self.options.dynamic_arch
-        cmake.definitions["USE_THREAD"] = self.options.use_thread
+            # This checks explicit user-specified fortran compiler
+            if not self._fortran_compiler:
+                if Version(self.version) < "0.3.21":
+                    self.output.warning(
+                        "Building with LAPACK support requires a Fortran compiler.")
+                else:
+                    tc.variables["C_LAPACK"] = True
+                    tc.variables["NOFORTRAN"] = True
+                    self.output.info(
+                        "Building LAPACK without Fortran compiler")
+
+        tc.variables["BUILD_WITHOUT_LAPACK"] = not self.options.build_lapack
+        tc.variables["DYNAMIC_ARCH"] = self.options.dynamic_arch
+        tc.variables["USE_THREAD"] = self.options.use_thread
+        tc.variables["USE_OPENMP"] = self.options.use_openmp
 
         # Required for safe concurrent calls to OpenBLAS routines
-        cmake.definitions["USE_LOCKING"] = not self.options.use_thread
+        tc.variables["USE_LOCKING"] = not self.options.use_thread
 
-        cmake.definitions[
-            "MSVC_STATIC_CRT"
-        ] = False  # don't, may lie to consumer, /MD or /MT is managed by conan
+        # don't, may lie to consumer, /MD or /MT is managed by conan
+        tc.variables["MSVC_STATIC_CRT"] = False
 
-        # This is a workaround to add the libm dependency on linux,
-        # which is required to successfully compile on older gcc versions.
-        cmake.definitions["ANDROID"] = self.settings.os in ["Linux", "Android"]
+        # Env variable escape hatch for enabling AVX512
+        no_avx512 = environ.get("NO_AVX512")
+        if no_avx512 is None:
+            no_avx512 = True
+        tc.variables["NO_AVX512"] = no_avx512
```

Comment:
Is there a reason this one isn't an option on the package? Strikes me that this would modify the binary that's being produced, thus we would want an option to make sure this is reflected in the package id and control when the package id changes or doesn't?

Reply:
This was a deliberate attempt of a "simple" fix to create functional binary packages for consumer CPUs that do not have AVX-512, without having to create a new option. The build server might have AVX-512, which causes unusable binary packages, as reported in #12705.

We could/should add this as an option to be more transparent. However, there are also several other options that we might expose, e.g. `NO_AVX`, `NO_AVX2`, and probably others as well. These instruction sets are more commonly available, though, and is less important to expose.

Do you want me to add a new option, e.g. `with_avx512`, which defaults to `False`?

Reply:
Yeah I think an option makes the most sense here. This is just how this is modelled in Conan. Especially if this is causing unusable binaries

Reply:
I've added a new option as suggested.